experimental/PropertyDDS/packages/property-changeset/src/utils.ts

```diff
@@ -2,2261 +2,2097 @@
+                        delete operationCS[removalInformation.typeid];
+                    }
+                    if (_.isEmpty(CS[operation])) {
+                        delete CS[operation];
+                    }
+                }
             } else {
-                const operationCS = CS[operation];
-                var typeidCS = CS[operation][removalInformation.typeid];
+                let typeidCS = CS[removalInformation.typeid];
                 delete typeidCS[removalInformation.lastSegment];
                 if (_.isEmpty(typeidCS)) {
-                    delete operationCS[removalInformation.typeid];
-                }
-                if (_.isEmpty(CS[operation])) {
-                    delete CS[operation];
+                    delete CS[removalInformation.typeid];
                 }
             }
-        } else {
-            var typeidCS = CS[removalInformation.typeid];
-            delete typeidCS[removalInformation.lastSegment];
-            if (_.isEmpty(typeidCS)) {
-                delete CS[removalInformation.typeid];
-            }
         }
+        return rootChangeSet;
     }
-    return rootChangeSet;
-};
 
-/**
- * Converts an array of paths to the tree structured representation that is needed
- * as input for the function getChangesToTokenizedPaths.
- *
- * @param {String} in_paths - An array with paths
- * @return {Map} A tree structured representation of the tokenized paths that can be
- *     passed to getChangesToTokenizedPaths and getFilteredChangeSetByPaths.
- */
-Utils.convertPathArrayToTree = function(in_paths) {
-    in_paths = _.isArray(in_paths) ? in_paths : [in_paths];
-    const pathsToProcess = new Set(in_paths);
-
-    // create an array of arrays splitting by .
-    const tokenizedPaths = in_paths.map((path) => tokenizePathString(path));
-
-    // Create a tree representation of the paths that are passed as an input so that
-    // we can leverage getChangesToTokenizedPaths and only be notified on paths
-    // that we care about.
-    const pathsToObj = _.reduce(tokenizedPaths, function(memo, tokenizedPath) {
-        let obj = memo;
-        let path = "";
-        let segment;
-
-        for (let index = 0; index < tokenizedPath.length; index++) {
-            segment = tokenizedPath[index];
-
-            if (index === 0) {
-                path = quotePathSegmentIfNeeded(segment);
-            } else {
-                path += `.${quotePathSegmentIfNeeded(segment)}`;
-            }
+    /**
+     * Converts an array of paths to the tree structured representation that is needed
+     * as input for the function getChangesToTokenizedPaths.
+     *
+     * @param  in_paths - An array with paths
+     * @returns {Map} A tree structured representation of the tokenized paths that can be
+     *     passed to getChangesToTokenizedPaths and getFilteredChangeSetByPaths.
+     */
+    export function convertPathArrayToTree(in_paths: string[]): TokenizedPath {
```

Comment:
As mentioned above, the typename TokenizedPath might be a bit misleading here. It is more a PathTree than just a single tokenized path. The JSDoc comment still states that this is just a Map.